frontend/spec/features/locations_spec.rb

```diff
@@ -0,0 +1,510 @@
+    expect(element.text).to eq 'Location Created'
+  end
+
+  it 'allows locations to be edited and persists the temporary location' do
+    location = create(:location)
+
+    visit "locations/#{location.id}/edit"
+    fill_in 'location_room_', with: '111'
+    find('#location_temporary_question_').click
+    select 'Conservation', from: 'location_temporary_'
+
+    # Click on save
+    find('button', text: 'Save Location', match: :first).click
+    element = find('.alert.alert-success.with-hide-alert')
+    expect(element.text).to eq 'Location Saved'
+    expect(find('#location_temporary_').value).to eq 'conservation'
+
+    find('a.hide-alert').click
+    # Click on save
+    find('button', text: 'Save Location', match: :first).click
+    element = find('.alert.alert-success.with-hide-alert')
+    expect(element.text).to eq 'Location Saved'
+    expect(find('#location_temporary_').value).to eq 'conservation'
+  end
+
+  it 'lists the new location in the browse list' do
+    now = Time.now.to_i
+
+    click_on 'Create'
+    click_on 'Location'
+    click_on 'Single Location'
+
+    element = find('h2')
+    expect(element.text).to eq 'New Location Location'
+
+    fill_in 'location_building_', with: "Location Building #{now}"
+    fill_in 'location_floor_', with: '1'
+    fill_in 'location_room_', with: '1'
+    fill_in 'location_coordinate_1_label_', with: "Label 1 #{now}"
+    fill_in 'location_coordinate_1_indicator_', with: "Indicator 1 #{now}"
+
+    # Click on save
+    find('button', text: 'Save Location', match: :first).click
+    element = find('.alert.alert-success.with-hide-alert')
+    expect(element.text).to eq 'Location Created'
+
+    run_index_round
+
+    visit '/'
+    click_on 'Browse'
+    find('a', text: 'Locations', match: :first).click
+
+    element = find('#filter-text')
+    element.fill_in with: "Location Building #{now}"
+    find('.sidebar input.text-filter-field + div button').click
+    expect(page).to have_xpath("//tr[contains(., 'Location Building #{now}')]")
+  end
+
+  it 'saves a valid repository-owned location' do
+    now = Time.now.to_i
+
+    click_on 'Create'
+    click_on 'Location'
+    click_on 'Single Location'
+    element = find('h2')
+    expect(element.text).to eq 'New Location Location'
+    fill_in 'location_building_', with: "Repository Building #{now}"
+    fill_in 'location_coordinate_1_label_', with: "Coordinate Label #{now}"
+    fill_in 'location_coordinate_1_indicator_', with: "Indicator Label #{now}"
+
+    element = find('#token-input-location_owner_repo__ref_')
+    element.fill_in with: 'locations_test_'
+    dropdown_items = all('li.token-input-dropdown-item2')
+    dropdown_items.first.click
+
+    # Click on save
+    find('button', text: 'Save Location', match: :first).click
+    element = find('.alert.alert-success.with-hide-alert')
+    expect(element.text).to eq 'Location Created'
+  end
+
+  it 'allows the new location to be viewed in non-edit mode' do
```

Comment:
@thimios I wonder why the original spec here clicks on 'Edit' to view the record, ported to your line 188 below, when the spec description above seems to intend viewing the record in _non_-edit mode?